src/akkudoktoreos/class_strompreis.py

```diff
@@ -22,51 +24,56 @@ def repeat_to_shape(array, target_shape):
 
 
 class HourlyElectricityPriceForecast:
-    def __init__(self, source, cache_dir="cache", charges=0.000228, prediction_hours=24):  # 228
-        self.cache_dir = cache_dir
-        os.makedirs(self.cache_dir, exist_ok=True)
-        self.cache_time_file = os.path.join(self.cache_dir, "cache_timestamp.txt")
-        self.prices = self.load_data(source)
+    def __init__(self, source: str | Path, config: AppConfig, charges=0.000228):  # 228
+        self._cache_dir = config.working_dir / config.directories.cache
+        if not self._cache_dir.is_dir():
```

Comment:
Since we create the directories at app startup, I would omit this check.

Reply:
the class could be used on it's own.That's why I put this early fail check

Reply:
I would argue since this is a server application after all, we can probably assume that the server startup is the only entrance point.